src/games/strategy/ui/SwingComponents.java

```diff
@@ -21,6 +21,7 @@
 public class SwingComponents {
 
   public enum KeyboardCode {
+    D(KeyEvent.VK_D),
     G(KeyEvent.VK_G);
```

Comment:
Yes! It'll grow as necessary.

Key benefits:
- we avoid the magic number interface swing uses. This class is evil: https://docs.oracle.com/javase/7/docs/api/javax/swing/SwingConstants.html, a number of swing interfaces are take multiple integer args, where each is a magic number. You always wonder, is it a magic number from the SwingConstants class? A subclass, or from the class you are using proper? 

So the enum gives us a strongly typed value, and we can avoid APIs that take in multiple int values, where each int means something different.

Second, we get trace ability. Doing a call hierarchy search on these reveals all usages. We'll be able to more easily answer the question of what shares which hot key. This also puts us on a path to replacing all of the specific keys with a user setting (namely buy consolidating usages and references to the keyboard keys, can eventually be more easily swapped out later)